src/utils/csv.rs

```diff
@@ -22,3 +21,15 @@ pub fn dump_data_to_csv<T: Serialize>(data: &[T], file_path: &Path) -> Result<()
     }
     Ok(())
 }
+
+pub fn dump_to_csv_or_print<T: serde::Serialize + Debug>(
+    output: Option<PathBuf>,
+    result: &[T],
```

Comment:
better to use an iterator here so that we don't have to use stuff like `map_as_vec`